scripts/version.sh

```
@@ -0,0 +1,229 @@
+        log_error "Invalid version format: $version (expected: MAJOR.MINOR.PATCH)"
+        exit 1
+    fi
+}
+
+# Increment version
+increment_version() {
+    local current_version=$1
+    local increment_type=$2
+    
+    local version_parts
+    version_parts=($(parse_version "$current_version"))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)